tracks/python/exercises/kindergarten-garden/mentoring.md

```diff
@@ -0,0 +1,36 @@
+### Reasonable solutions
+
+```python
+class Garden(object):
+
+    DEFAULT_STUDENTS = [
+        'Alice', 'Bob', 'Charlie', 'David',
+        'Eve', 'Fred', 'Ginny', 'Harriet',
+        'Ileana', 'Joseph', 'Kincaid', 'Larry',
+    ]
+
+    PLANT_NAMES = {
+        'C' : 'Clover',
+        'G' : 'Grass',
+        'R' : 'Radishes',
+        'V' : 'Violets'
+    }
+
+    def __init__(self, diagram, students=DEFAULT_STUDENTS):
+        self.rows = diagram.split('\n')
+        self.students = sorted(students)
+
+    def plants(self, student):
+        student_index = self.students.index(student)
+        row_index = student_index * 2
+        letters = ''.join(row[row_index : row_index + 2] for row in self.rows)
+        return [self.PLANT_NAMES[letter] for letter in letters]
+```
+
+
+### Common suggestions
+- Using a constant dict to map from letters to plant names, like `PLANT_NAMES` above.
+
+### Talking points
+- List comprehensions for lists of two things aren't that much shorter that just two copies of the code, but they work if the problem expands to more rows.
+- You can build the assignments of plants to students once, in the constructor, on on-demand in the `plants()` method.
```

Comment:
L36 should probably read "or on-demand in the `Garden.plants` method" ... that said, I'm not sure we should encourage that as the default approach. I'd personally always rather see a solution that uses a **dict** to map each student to their list of plant names; it avoids the costly **list.index** call for every student, and it means the parsing happens only once for all students and never twice (or more) for the same student.